firrtl/src/main/scala/firrtl/ir/Serializer.scala

```diff
@@ -337,15 +337,23 @@ object Serializer {
     case Field(name, flip, tpe) => s(flip); b ++= legalize(name); b ++= " : "; s(tpe)
   }
 
-  private def s(node: Type)(implicit b: StringBuilder, indent: Int): Unit = node match {
+  private def s(node: Type)(implicit b: StringBuilder, indent: Int): Unit = s(node, false)
+
+  private def s(node: Type, lastEmittedConst: Boolean)(implicit b: StringBuilder, indent: Int): Unit = node match {
     // Types
     case ProbeType(underlying: Type) => b ++= "Probe<"; s(underlying); b += '>'
     case RWProbeType(underlying: Type) => b ++= "RWProbe<"; s(underlying); b += '>'
-    case ConstType(underlying: Type) => b ++= "const "; s(underlying)
+    case ConstType(underlying: Type) => {
+      // Avoid emitting multiple consecurive 'const', which can otherwise occur for const vectors of const elements
+      if (!lastEmittedConst) {
+        b ++= "const "
+      }
+      s(underlying, true)(b, indent)
+    }
```

Comment:
I'm of two minds about this solution. I'm a bit concerned because this same issue is solved for directions in the Converter (https://github.com/chipsalliance/chisel/blob/a6728aa2e81fd96824415693d9710165c50f90cb/core/src/main/scala/chisel3/internal/firrtl/Converter.scala#L321), and presumably we don't have it solved for probes. I'm worried that when we fold the Converter and Serializer together, we will lose this fix. I grant there is some elegance to the spot in that it ensures that any FIRRTL IR will be emitted properly whereas the Converter fix doesn't guarantee that. But we don't create FIRRTL IR any other way so I think I prefer the further upstream fix.

Better yet would be if we could pull the fix further up into the actual construction of the Chisel types such that these fixups aren't necessary.